05_debug/sysfs/debug_sysfs.c

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/printk.h>
+#include <linux/device.h>
+#include <linux/string.h>
+
+MODULE_AUTHOR("Oleh Yakymenko <oleh.o.yakymenko@globallogic.com");
+MODULE_DESCRIPTION("Debug sysfs to lowercase converter");
+MODULE_LICENSE("Dual BSD/GPL");
+MODULE_VERSION("0.1");
+
+static struct class *class_dsysfs;
+static struct device *device_dsysfs;
+static dev_t dev_t_dsysfs;
+
+static char s[PAGE_SIZE] = "string empty!";
+static int total_calls_show;
+static int total_calls_write;
+static int total_char_processed;
+static int total_char_converted;
+
+
+ssize_t show_lowercase(struct device *dev, struct device_attribute *attr,
+			char *buf)
+{
+	total_calls_show++;
+	return sprintf(buf, "%s\n", s);
+}
+
+ssize_t store_lowercase(struct device *dev, struct device_attribute *attr,
+			 const char *buf, size_t count)
+{
+	int s_offset = 0;
+	int retval = 0;
+
+	total_calls_write++;
+
+	retval = sscanf(buf, "%s\n", s);
+
+	if (retval < 0) {
+		pr_err("Can't copy your string\n");
+		return count;
+	}
+
+	pr_info("Before convertion = %s\n", s);
+
+	for (; s[s_offset] != '\0'; s_offset++) {
+		if (s[s_offset] >= 'A' && s[s_offset] <= 'Z') {
+			s[s_offset] = s[s_offset] + ('a' - 'A');
```

Comment:
operator += looks better
it is possible to use pointer to char instead of indexing with offset.